gateway/src/apicast/configuration/service.lua

```diff
@@ -30,13 +33,42 @@ local http_methods_with_body = {
   PATCH = true
 }
 
+local function is_http2_request()
+  return ngx.req.http_version == '2.0'
+end
+
+local function content_type_is_urlencoded(headers)
+    local ct = headers["Content-Type"]
+    if not ct then
+        return false
+    end
+
+    -- Handle duplicate headers
+    -- This shouldn't happen but can in the real world
+    if type(ct) ~= "string" then
+        ct = tbl_concat(ct, ",")
+    end
+
+    return str_find(lower(ct), "application/x-www-form-urlencoded", 1, true) ~= nil
+end
+
+
 local function read_body_args(...)
   local method = ngx.req.get_method()
 
+  ngx.log(ngx.WARN, "is_http2_request: ", is_http2_request())
```

Comment:
I would move this log inside the if statement, otherwise this will print every time this function is called.

Reply:
good catch! I forgot to remove it